File: src/main/java/io/rakam/shareablereports/io/rakam/shareablereports/model/Report.java
package io.rakam.shareablereports.io.rakam.shareablereports.model;

public class Report {

    private final String reportName;
    private final String reportTitle;
    private final String reportDescription;
    private final String reportAuthor;

    public Report(String reportName, String reportTitle, String reportDescription, String reportAuthor) {
        this.reportName = reportName;
        this.reportTitle = reportTitle;
        this.reportDescription = reportDescription;
        this.reportAuthor = reportAuthor;
    }

    public String getReportName() {
        return reportName;
    }

    public String getReportTitle() {
        return reportTitle;
    }

    public String getReportDescription() {
        return reportDescription;
    }

    public String getReportAuthor() {
        return reportAuthor;
    }

    @Override
    public String toString() {
        final StringBuffer sb = new StringBuffer("Report{");
        sb.append("reportName='").append(reportName).append('\'');
        sb.append(", reportTitle='").append(reportTitle).append('\'');
        sb.append(", reportDescription='").append(reportDescription).append('\'');
        sb.append(", reportAuthor='").append(reportAuthor).append('\'');
        sb.append('}');
        return sb.toString();
    }
}

File: src/main/java/io/rakam/shareablereports/entity/Report.kt
package io.rakam.shareablereports.entity

class Report {
    private var id: Long? = null
    private var reportTitle: String? = null
    private var reportDescription: String? = null
    private var reportAuthor: String? = null

    constructor()

    constructor(id: Long?, reportTitle: String?, reportDescription: String?, reportAuthor: String?) {
        this.id = id
        this.reportTitle = reportTitle
        this.reportDescription = reportDescription
        this.reportAuthor = reportAuthor
    }

    fun getId(): Long? {
        return id
    }

    fun setId(id: Long) {
        this.id = id
    }

    fun getReportTitle(): String? {
        return reportTitle
    }

    fun setReportTitle(reportTitle: String) {
        this.reportTitle = reportTitle
    }

    fun getReportDescription(): String? {
        return reportDescription
    }

    fun setReportDescription(reportDescription: String?) {
        this.reportDescription = reportDescription
    }

    fun getReportAuthor(): String? {
        return reportAuthor
    }

    fun setReportAuthor(reportAuthor: String?) {
        this.reportAuthor = reportAuthor
    }

    override fun toString(): String {
        return "Report(id=$id, reportTitle=$reportTitle, reportDescription=$reportDescription, reportAuthor=$reportAuthor)"
    }
}